src/devices/llama_devices.ts

```diff
@@ -0,0 +1,169 @@
+            value: 0,
+        },
+        {
+            name: 'off/on',
+            value: 1,
+        },
+        {
+            name: 'toggle',
+            value: 2,
+        },
+    ],
+};
+const configs: Config[] = [
+    switchTypeConf,
+    switchAtionConf,
+];
+
+function addEndpointPrefix(ep: number) {
+    return `${prefixEnpointName}${ep}`;
+}
+
+function createSwitchExposes(epc: number) {
+    const feature: exposes.Enum[] = [];
+    for (let i = 0; i < epc; i++) {
+        configs.forEach((c) => {
+            feature.push(e.enum(c.name, ea.ALL,
+                c.values.map((v) => v.name))
+                .withDescription(c.description)
+                .withEndpoint(addEndpointPrefix(i + 1),
+                ));
+        });
+    }
+    return feature;
+}
+
+function mapObject<T>(arr: T[], mapKey: (k: T) => string | [string], mapValue: (v: T) => string | number) {
+    return Object.fromEntries(arr.map((el, i, arr) => [mapKey(el), mapValue(el)]));
+}
+
+const fzLocal = {
+    on_off_switch_config: {
+        cluster: 'genOnOffSwitchCfg',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            return mapObject(
+                configs,
+                (k) => [`${k.name}_${addEndpointPrefix(msg.endpoint.ID)}`],
+                (v) => {
+                    const vv = v.values.find((vv) => vv.value == msg.data[v.attribute]);
+                    return vv?.name;
+                });
+        },
+    } satisfies Fz.Converter,
+    on_off_action: {
+        cluster: 'genOnOff',
+        type: [
+            'commandOn',
+            'commandOff',
+            'commandToggle',
+            // 'commandOnWithTimedOff'
+        ],
+        convert: (model, msg, publish, options, meta) => {
+            return {action: `${msg.endpoint.ID}_${msg.type.replace('command', '')}`};
+        },
+    } satisfies Fz.Converter,
+};
+
+const tzLocal = {
+    on_off_switch_config: {
+        key: ['switch_type', 'switch_actions'],
+        convertGet: async (entity, key, meta) => {
+            await entity.read('genOnOffSwitchCfg', configs.map((c) => c.attribute));
+        },
+        convertSet: async (entity, key, value, meta) => {
+            const cfg = configs.find((c) => c.name == key);
+            const ret = mapObject([cfg], (k) => k.attribute, (v) => v.values.find((v) => v.name == value).value);
+            await entity.write('genOnOffSwitchCfg', ret);
+
+            return {state: {[key]: value}};
+        },
+    } satisfies Tz.Converter,
+};
+
+const definitions: Definition[] = [
+    {
+        zigbeeModel: ['Zucchini'],
+        model: 'Zucchini',
+        vendor: 'Llama Devices',
+        description: 'On/Off output device',
+        fromZigbee: [fzLocal.on_off_action, fz.battery, fz.temperature, fz.humidity, fzLocal.on_off_switch_config],
+        toZigbee: [tzLocal.on_off_switch_config],
+        exposes: [
+            e.action([]),
```

Comment:
There are no actions here, it can be removed?

Reply:
value for actions generates in function, not in exposes. anyway, I`ll use extend

Reply:
But still actions need to be filled in here.

Reply:
for what purpose? in this place they have no effect at all

Reply:
done